src/main/kotlin/org/team5419/fault/BerkeliumIterativeRobotBase.kt

```diff
@@ -0,0 +1,139 @@
+
+    // periodic methods
+    open fun robotPeriodic() {
+        if (mRobotFirstLoop) {
+            println("Default robot periodic method... Override me")
+            mRobotFirstLoop = false
+        }
+    }
+
+    open fun disabledPeriodic() {
+        if (mDisabledFirstLoop) {
+            println("Default disabled periodic method... Override me")
+            mDisabledFirstLoop = false
+        }
+    }
+
+    open fun teleopPeriodic() {
+        if (mTeleopFirstLoop) {
+            println("Default teleop periodic method... Override me")
+            mTeleopFirstLoop = false
+        }
+    }
+
+    open fun autonomousPeriodic() {
+        if (mAutoFirstLoop) {
+            println("Default autonomous periodic method... Override me")
+            mAutoFirstLoop = false
+        }
+    }
+
+    open fun testPeriodic() {
+        if (mTestFirstLoop) {
+            println("Default test periodic method... Override me")
+            mTestFirstLoop = false
+        }
+    }
+
+    // loop func
+    @Suppress("ComplexMethod")
+    protected fun loopFunc() {
+        mLoopStartTime = Timer.getFPGATimestamp().seconds
+        if (isDisabled) {
+            if (mLastMode != Mode.Disabled) {
+//                Shuffleboard.disableActuatorWidgets()
+                disabledInit()
+                mLastMode = Mode.Disabled
+            }
+            HAL.observeUserProgramDisabled()
+            disabledPeriodic()
+        } else if (isAutonomous) {
+            if (mLastMode != Mode.Autonomous) {
+//              Shuffleboard.disableActuatorWidgets()
+                autonomousInit()
+                mLastMode = Mode.Autonomous
+            }
+            HAL.observeUserProgramAutonomous()
+            autonomousPeriodic()
+        } else if (isOperatorControl) {
+            if (mLastMode != Mode.Teleop) {
+                teleopInit()
+                mLastMode = Mode.Teleop
+            }
+            HAL.observeUserProgramTeleop()
+            teleopPeriodic()
+        } else {
+            if (mLastMode != Mode.Test) {
+                testInit()
+                mLastMode = Mode.Test
+            }
+            HAL.observeUserProgramTest()
+            testPeriodic()
+        }
+        robotPeriodic()
+        mLoopEndTime = Timer.getFPGATimestamp().seconds
+        // check for overrun
+        if (mLoopEndTime > (mPeriod + mLoopStartTime)) {
+            // overrun
+            mOverrunCount += 1
+        }
+        // check if we should print overruns
+        if ((mLoopEndTime - mLastOverrunDump) > mOverrunPrintPeriod) {
+            // check if we have overruns
+            if (mOverrunCount > 0) {
+                println("WARNING: There have been $mOverrunCount loop overruns in the " +
+                        "past ${mLoopEndTime - mLastOverrunDump} seconds!")
+
+                // set overruns to 0
+                mOverrunCount = 0
+            }
+            mLastOverrunDump = mLoopEndTime
+        }
```

Comment:
comments like `set overruns to 0` aren't really useful - comments about the overall purpose of the algorithm are though. ie what even is an overrun and why do we care? what is the overall method for detecting and reporting them?